### PR TITLE
feat: scratch org async/resume

### PR DESCRIPTION
**Pull request by mshanemc**
requires https://github.com/forcedotcom/sfdx-core/pull/551
### What does this PR do?
--async on scratch create
resume command

### What issues does this PR fix or reference?
[@W-10657657@](https://gus.lightning.force.com/a07EE00000m2WwPYAU)

### CODE REVIEW

**Review by peternhale — CHANGES_REQUESTED** [review]

File: messages/create_scratch.md [hunk 1]
```diff
@@ -87,3 +91,11 @@ OAuth client secret of your personal connected app
 # success
 
 Your scratch org is ready.
+
+# timeoutUnexpected
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply messages/create_scratch.md:95]: looks like I didn't use it at all (well, technically it moved to core.  so deleting)

File: src/commands/env/create/scratch.ts [hunk 2]
```diff
@@ -128,46 +106,64 @@ export default class EnvCreateScratch extends SfCommand<ScratchCreateResponse> {
...
+
+      return { username, scratchOrgInfo, authFields, warnings, orgId: scratchOrgInfo.Id };
+    } catch (error) {
+      if (error instanceof SfError && error.name === 'ScratchOrgInfoTimeoutError') {
+        this.spinner.stop(lastStatus);
+        const scratchOrgInfoId = (error.data as { scratchOrgInfoId: string }).scratchOrgInfoId;
+        const resumeMessage = messages.getMessage('action.resume', [scratchOrgInfoId]);
+
+        this.log(resumeMessage);
+        this.error('The scratch org did not complete within your wait time', { code: '69', exit: 69 });
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/commands/env/create/scratch.ts:162]: Huh....you may have some code missing in your sandbox PR.  I see this one as an SfError https://github.com/salesforcecli/plugin-env/blob/8c80ebc35ed0bec25a42b359dfd693a81eb6ff1d/src/shared/sandboxCommandBase.ts#L140

that you catch and throw at

https://github.com/salesforcecli/plugin-env/blob/8c80ebc35ed0bec25a42b359dfd693a81eb6ff1d/src/commands/env/create/sandbox.ts#L192

FWIW I don't expect my scratch orgs to time out.  It's great that it's not a complete brick now thanks to `resume`, but if I didn't say `--async` I expect it to finish.

[the remark this answers is not in the document]
Comment (peternhale) [reply src/commands/env/create/scratch.ts:162]: The revelation of how I treat timeouts has yet to be pushed to git. That handle function no longer exists. But I understand that most scratch org creates finished within the default wait.

**Review by peternhale — APPROVED** [review]

**Review by peternhale — DISMISSED** [review]

File: src/commands/env/create/scratch.ts [hunk 3]
```diff
@@ -125,49 +103,66 @@ export default class EnvCreateScratch extends SfCommand<ScratchCreateResponse> {
...
+        lastStatus = buildStatus(data, baseUrl);
+        this.spinner.status = lastStatus;
+      });
+    }
     this.log();
-    this.spinner.start('Creating Scratch Org');
-    const { username, scratchOrgInfo, authFields, warnings } = await flags['target-dev-hub'].scratchOrgCreate(
-      createCommandOptions
+    this.spinner.start(
+      flags.async ? 'Requesting Scratch Org (will not wait for completion because --async)' : 'Creating Scratch Org'
```
Comment (RodEsp) [inline src/commands/env/create/scratch.ts:139]: @jshackell-sfdc could you review this output too please?

[the remark this answers is not in the document]
Comment (jshackell-sfdc) [reply src/commands/env/create/scratch.ts:139]: Okay?  Made the message a bit longer, but hopefully that's still okay.

**Pull request closed without merging**